### PR TITLE
Generalize Modal components to improve flexibility

### PR DESCRIPTION
**Pull request by addisonhernandez**
@Acid-Override @akb3y @positivefx 
I'm refactoring the modal components into a more general form.

Here's the current interface. Input is welcome!

### Props:

 - `showModal` -- Boolean value that determines whether or not to render this component
 - `onClose` -- Function which when called (`onClose()`) will call `setShowModal` and toggle the state of `showModal`
 - `ModalForm` -- React component which renders whatever content you want to display in the modal
 - `formData` -- JavaScript Object that the Modal component will pass to ModalForm. This is how you can pass props to your ModalForm!

### CODE REVIEW

File: client/src/components/QuestionAnswers/QAList.jsx [hunk 1]
```diff
@@ -65,9 +66,10 @@ function QAList() {
         ) : null}
         <button type="button" onClick={() => setShowAddQuestion(true)}>Add a question</button>
       </div>
-      <AddQuestionModal
+      <Modal
```
Comment (positivefx) [inline client/src/components/QuestionAnswers/QAList.jsx:69]: nice modal wrapper

**Review by positivefx — APPROVED** [review]

File: client/src/components/QuestionAnswers/Question.jsx [hunk 2]
```diff
@@ -61,10 +62,11 @@ function Question({ questionId, questionBody, questionHelpfulness }) {
         </span>
       </div>
       {/* TODO: is this the best place for the modal? */}
-      <AddAnswerModal
+      <Modal
         showModal={showAddAnswer}
         onClose={() => setShowAddAnswer(false)}
-        productData={{ productName, questionId, questionBody }}
+        ModalForm={AddAnswerModal}
+        formData={{ productName, questionId, questionBody }}
```
Comment (Acid-Override) [inline client/src/components/QuestionAnswers/Question.jsx:69]: ModalForm and formData !  Cool... I need to look into this for my comparison.

File: client/src/components/modals/AddAnswerModal.jsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          name="your-email"
+          maxLength={60}
+          placeholder="Example: jack@email.com"
+          value={newAnswerEmail}
+          onChange={(e) => setNewAnswerEmail(e.target.value)}
+          required
+        />
+      </label>
+      <span style={{ fontSize: '.87em' }}>
+        <em>For authentication reasons, you will not be emailed</em>
```
Comment (Acid-Override) [inline client/src/components/modals/AddAnswerModal.jsx:88]: Nice user message :P

File: client/src/components/QuestionAnswers/Question.jsx (hunk 2 above)
Comment (addisonhernandez) [reply client/src/components/QuestionAnswers/Question.jsx:69]: Yeah!

Interestingly, this is a Higher-Order Component, like the one that the change request is looking for

File: client/src/components/modals/AddAnswerModal.jsx (hunk 3 above)
Comment (addisonhernandez) [reply client/src/components/modals/AddAnswerModal.jsx:88]: Lol, it's a requirement from the BRD. It doesn't make sense to me

**Comment by addisonhernandez** [comment]
[Generalize Modal Dialog](https://trello.com/c/IORpYbgK/36-generalize-modal-dialog)

**Pull request merged**